### PR TITLE
Spring Ball can now be bound to any items or no items

### PR DESCRIPTION
**Pull request by UltiNaruto**
# Spring Ball can now be bound to any items or no items

If springBallItem is set to Nothing it requires nothing to use Spring Ball.
If springBallItem is not set it defaults to bombs.
springBallItem cannot have Ice Trap or Floaty Jump as values since they are temp pickups.

### CODE REVIEW

File: src/patch_config.rs [hunk 1]
```diff
@@ -2428,6 +2436,21 @@ impl PatchConfigPrivate {
...
+                .game_config
+                .spring_ball_item
+                .as_deref()
+            {
+                Some(s) => PickupType::from_str(s),
+                None => PickupType::MorphBallBomb
+            }
+        };
+
+        if [PickupType::FloatyJump, PickupType::IceTrap].contains(&spring_ball_item) {
```
Comment (toasterparty) [inline src/patch_config.rs:2450]: This code would be forwards compatible if you did:

```rs
if spring_ball_item as u32 > PickupType::Nothing as u32
```

**Review by toasterparty — CHANGES_REQUESTED** [review]
Looks good, just one minor detail

File: src/patch_config.rs (hunk 1 above)
Comment (toasterparty) [reply src/patch_config.rs:2450]: Nothing is `41`, which is beyond the highest real pickup id (`Newborn = 40`)

Additional pickup types get added like so:

```
Newborn = 40
Nothing = 41
FloatyJump = 42
IceTrap = 43
etc....
```

Comment (UltiNaruto) [reply src/patch_config.rs:2450]: Oh I see

**Pull request merged**